src/segment_az.go

```diff
@@ -36,24 +36,47 @@ func (a *az) init(props *properties, env environmentInfo) {
 }
 
 func (a *az) enabled() bool {
```

Comment:
I decided to make the EnvVars have priority over the CLI for two reasons:
1. The CLI is slow and this segment has always been slow on my computer.  So we do the faster operation first, and early exit if it was successful.
2. The EnvVars is something the user is opting in to by adding code to the $profile.  To me, the act of doing that signals that the user wants the EnvVars to be used.